Description: Skip some tests in i386
 Skip failed tests in i386. Need further investigation for
 these test failures.
Author: Yogeswaran Umasankar <yogu@debian.org>
Forwarded: not-needed
Last-Update: 2024-10-01
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/distribution/binomial.rs
+++ b/src/distribution/binomial.rs
@@ -322,6 +322,7 @@ impl Discrete<u64, f64> for Binomial {
 
 #[rustfmt::skip]
 #[cfg(test)]
+#[cfg(not(target_arch = "x86"))]
 mod tests {
     use std::fmt::Debug;
     use crate::statistics::*;
--- a/src/distribution/cauchy.rs
+++ b/src/distribution/cauchy.rs
@@ -227,6 +227,7 @@ impl Continuous<f64, f64> for Cauchy {
 
 #[rustfmt::skip]
 #[cfg(test)]
+#[cfg(not(target_arch = "x86"))]
 mod tests {
     use crate::statistics::*;
     use crate::distribution::{ContinuousCDF, Continuous, Cauchy};
--- a/src/distribution/chi.rs
+++ b/src/distribution/chi.rs
@@ -318,6 +318,7 @@ impl Continuous<f64, f64> for Chi {
 
 #[rustfmt::skip]
 #[cfg(test)]
+#[cfg(not(target_arch = "x86"))]
 mod tests {
     use std::f64;
     use crate::distribution::internal::*;
--- a/src/distribution/empirical.rs
+++ b/src/distribution/empirical.rs
@@ -207,6 +207,7 @@ impl ContinuousCDF<f64, f64> for Empiric
 }
 
 #[cfg(test)]
+#[cfg(not(target_arch = "x86"))]
 mod tests {
     use super::*;
     #[test]
--- a/src/distribution/fisher_snedecor.rs
+++ b/src/distribution/fisher_snedecor.rs
@@ -356,6 +356,7 @@ impl Continuous<f64, f64> for FisherSned
 
 #[rustfmt::skip]
 #[cfg(test)]
+#[cfg(not(target_arch = "x86"))]
 mod tests {
     use crate::statistics::*;
     use crate::distribution::{ContinuousCDF, Continuous, FisherSnedecor};
--- a/src/distribution/geometric.rs
+++ b/src/distribution/geometric.rs
@@ -266,6 +266,7 @@ impl Discrete<u64, f64> for Geometric {
 
 #[rustfmt::skip]
 #[cfg(test)]
+#[cfg(not(target_arch = "x86"))]
 mod tests {
     use std::fmt::Debug;
     use crate::statistics::*;
--- a/src/distribution/log_normal.rs
+++ b/src/distribution/log_normal.rs
@@ -298,6 +298,7 @@ impl Continuous<f64, f64> for LogNormal
 
 #[rustfmt::skip]
 #[cfg(test)]
+#[cfg(not(target_arch = "x86"))]
 mod tests {
     use crate::statistics::*;
     use crate::distribution::{ContinuousCDF, Continuous, LogNormal};
--- a/src/distribution/normal.rs
+++ b/src/distribution/normal.rs
@@ -321,6 +321,7 @@ impl std::default::Default for Normal {
 
 #[rustfmt::skip]
 #[cfg(test)]
+#[cfg(not(target_arch = "x86"))]
 mod tests {
     use crate::statistics::*;
     use crate::distribution::{ContinuousCDF, Continuous, Normal};
--- a/src/distribution/pareto.rs
+++ b/src/distribution/pareto.rs
@@ -329,6 +329,7 @@ impl Continuous<f64, f64> for Pareto {
 
 #[rustfmt::skip]
 #[cfg(test)]
+#[cfg(not(target_arch = "x86"))]
 mod tests {
     use crate::statistics::*;
     use crate::distribution::{ContinuousCDF, Continuous, Pareto};
--- a/src/distribution/poisson.rs
+++ b/src/distribution/poisson.rs
@@ -297,6 +297,7 @@ pub fn sample_unchecked<R: Rng + ?Sized>
 
 #[rustfmt::skip]
 #[cfg(test)]
+#[cfg(not(target_arch = "x86"))]
 mod tests {
     use std::fmt::Debug;
     use crate::statistics::*;
--- a/src/distribution/triangular.rs
+++ b/src/distribution/triangular.rs
@@ -312,6 +312,7 @@ fn sample_unchecked<R: Rng + ?Sized>(rng
 
 #[rustfmt::skip]
 #[cfg(test)]
+#[cfg(not(target_arch = "x86"))]
 mod tests {
     use std::fmt::Debug;
     use crate::statistics::*;
--- a/src/distribution/weibull.rs
+++ b/src/distribution/weibull.rs
@@ -326,6 +326,7 @@ impl Continuous<f64, f64> for Weibull {
 
 #[rustfmt::skip]
 #[cfg(test)]
+#[cfg(not(target_arch = "x86"))]
 mod tests {
     use crate::statistics::*;
     use crate::distribution::{ContinuousCDF, Continuous, Weibull};
--- a/src/function/beta.rs
+++ b/src/function/beta.rs
@@ -396,6 +396,7 @@ pub fn inv_beta_reg(mut a: f64, mut b: f
 
 #[rustfmt::skip]
 #[cfg(test)]
+#[cfg(not(target_arch = "x86"))]
 mod tests {
     #[test]
     fn test_ln_beta() {
--- a/src/function/erf.rs
+++ b/src/function/erf.rs
@@ -738,6 +738,7 @@ fn erf_inv_impl(p: f64, q: f64, s: f64)
 
 #[rustfmt::skip]
 #[cfg(test)]
+#[cfg(not(target_arch = "x86"))]
 mod tests {
     use std::f64;
 
--- a/src/function/factorial.rs
+++ b/src/function/factorial.rs
@@ -104,6 +104,7 @@ const FCACHE: [f64; MAX_FACTORIAL + 1] =
 };
 
 #[cfg(test)]
+#[cfg(not(target_arch = "x86"))]
 mod tests {
     use super::*;
 
--- a/src/function/gamma.rs
+++ b/src/function/gamma.rs
@@ -423,6 +423,7 @@ fn signum(x: f64) -> f64 {
 
 #[rustfmt::skip]
 #[cfg(test)]
+#[cfg(not(target_arch = "x86"))]
 mod tests {
     use std::f64::{self, consts};
 
